File: chroma-league/src/main/java/com/bonepl/chromaleague/state/RunningState.java
package com.bonepl.chromaleague.state;

import com.bonepl.chromaleague.tasks.ChangeAwareBoolean;

public final class RunningState {
    private static final ChangeAwareBoolean RIOT_API = new ChangeAwareBoolean();
    private static final ChangeAwareBoolean RUNNING_GAME = new ChangeAwareBoolean();
    private static GameState gameState = null;

    private RunningState() {
    }

    public static ChangeAwareBoolean getRiotApi() {
        return RIOT_API;
    }

    public static ChangeAwareBoolean getRunningGame() {
        return RUNNING_GAME;
    }

    public static void setRunningGame(boolean newValue) {
        if (RUNNING_GAME.different(newValue)) {
            if (newValue) {
                gameState = new GameState();
            } else {
                gameState = null;
            }
        }
        RUNNING_GAME.setValue(newValue);
    }

    public static void setRiotApi(boolean newValue) {
        RIOT_API.setValue(newValue);
    }

    public static GameState getGameState() {
        return gameState;
    }
}

File: src/state/RunningState.ts
import {ChangeAwareBoolean} from '../tasks/ChangeAwareBoolean.js';
import {GameState} from './GameState.js';

export class RunningState {
    private static riotApi = new ChangeAwareBoolean();
    private static runningGame = new ChangeAwareBoolean();
    private static gameState = new GameState();
    static getRiotApi(): ChangeAwareBoolean {
        return this.riotApi;
    }

    static setRiotApi(value: boolean): void {
        this.riotApi.setValue(value);
        if (!value) {
            this.runningGame.reset();
        }
    }

    static getRunningGame(): ChangeAwareBoolean {
        return this.runningGame;
    }

    static setRunningGame(value: boolean): void {
        if (this.runningGame.different(value)) {
            this.gameState = new GameState();
        }
        this.runningGame.setValue(value);
    }

    static getGameState(): GameState {
        return this.gameState;
    }
}
